gulp/plugins/gulp-example-source.js

```diff
@@ -4,51 +4,38 @@ import path from 'path'
 import through from 'through2'
 import Vinyl from 'vinyl'
 
-// Heads up!
-// This plugin is not universal, so it's okay to keep all existing sources as the state.
-// https://github.com/Semantic-Org/Semantic-UI-React/issues/3095
-const exampleSources = {}
 const pluginName = 'gulp-example-source'
 
 export default () => {
-  function bufferContents(file, enc, cb) {
-    if (file.isNull()) {
-      cb(null, file)
-      return
-    }
+  const exampleSources = {}
 
-    if (file.isStream()) {
-      cb(new gutil.PluginError(pluginName, 'Streaming is not supported'))
-      return
-    }
+  return through.obj(
+    (file, enc, cb) => {
+      if (file.isNull()) {
+        cb(null, file)
+        return
+      }
 
-    try {
-      const sourceName = _
-        .split(file.path, path.sep)
+      if (file.isStream()) {
+        cb(new gutil.PluginError(pluginName, 'Streaming is not supported'))
+        return
+      }
+
+      const sourceName = _.split(file.path, path.sep)
         .slice(-4)
         .join('/')
         .slice(0, -3)
 
       exampleSources[sourceName] = file.contents.toString()
```

Comment:
Also here, separate files would be great.

Reply:
Yep, for next iteration of this plugin it will be great.